src/github.com/getlantern/flashlight/analytics/analytics.go

```diff
@@ -73,25 +77,44 @@ func sessionVals(ip, version, clientId, sc string) string {
 	// Custom variable for the Lantern version
 	vals.Add("cd1", version)
 
+	// Custom dimension for the hash of the executable
+	hash := getExecutableHash()
+	vals.Add("cd2", hash)
+
 	// This forces the recording of the session duration. It must be either
 	// "start" or "end". See:
 	// https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
 	vals.Add("sc", sc)
 	return vals.Encode()
 }
 
-func endSession(ip string, version string, proxyAddrFN eventual.Getter, clientId string) {
-	args := sessionVals(ip, version, clientId, "end")
+// GetExecutableHash returns the hash of the currently running executable.
+// If there's an error getting the hash, this returns
+func getExecutableHash() string {
+	if lanternPath, err := osext.Executable(); err != nil {
```

Comment:
Good point. That analytics code doesn't actually run on Android but we
should guard against it.
On May 3, 2016 8:18 PM, "fffw" <notifications@github.com> wrote:

> In src/github.com/getlantern/flashlight/analytics/analytics.go
> <https://github.com/getlantern/lantern/pull/4199#discussion_r61986708>:
>
> >  	// This forces the recording of the session duration. It must be either
> >  	// "start" or "end". See:
> >  	// https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
> >  	vals.Add("sc", sc)
> >  	return vals.Encode()
> >  }
> >
> > -func endSession(ip string, version string, proxyAddrFN eventual.Getter, clientId string) {
> > -	args := sessionVals(ip, version, clientId, "end")
> > +// GetExecutableHash returns the hash of the currently running executable.
> > +// If there's an error getting the hash, this returns
> > +func getExecutableHash() string {
> > +	if lanternPath, err := osext.Executable(); err != nil {
>
> Oh actually Android is Linux, but I don't know what it will get because
> Lantern is not an executable on Android.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/getlantern/lantern/pull/4199/files/18d2e39a813e04901f2eda3361fdf383c732675b#r61986708>
>